File: algorithms/easy/number_of_senior_citizens.ex
defmodule Solution do
  @spec count_seniors(details :: [String.t()]) :: integer
  def count_seniors(details) do
    details
    |> Enum.reduce(0, fn d, acc ->
      age = d |> String.slice(11..12) |> String.to_integer()

      if age > 60 do
        acc + 1
      else
        acc
      end
    end)
  end
end

File: algorithms/easy/number_of_senior_citizens.rs
impl Solution {
    pub fn count_seniors(details: Vec<String>) -> i32 {
        let mut ans = 0;
        for d in details {
            let age = &d[11..=12].parse::<i32>().unwrap();
            if *age > 60 {
                ans += 1;
            }
        }
        ans
    }
}
